pkg/apis/etcd/v1alpha1/types.go

```diff
@@ -23,6 +23,9 @@ type EtcdStorageConditionType string
 const (
 	// Deployed means EtcdProxy ReplicaSet and Service for exposing EtcdProxy are created.
 	Deployed EtcdStorageConditionType = "Deployed"
+
+	// OwnerReferenceUpdated means an EtcdStorage-related resource is updated with EtcdStorage OwnerReference.
+	OwnerReferenceUpdated EtcdStorageConditionType = "OwnerReferenceUpdated"
```

Comment:
Decided to go with two conditions, because if we're updating the OwnerRef, that means the object exists, i.e. EtcdStorage is deployed and probably usable. For some reason it's not owned by EtcdProxy controller, and we can't update it (could be due to RBAC, cluster configuration, etc).

Reply:
Conditions should be end user focused. The end user does not care about your owner references.

The end user cares about whether the etcd proxy is up (Deployed) and whether certificates have been propagated.

Reply:
This totally makes sense. I've updated the PR to remove this condition.